src/getAverage.js

```diff
@@ -0,0 +1,34 @@
+'use strict';
+
+/**
+ * Implement a custom array method getAverage:
+ *
+ * Method works in different ways:
+ * - without arguments: calculates average value in array (sum of elements
+ * divided by array length)
+ *
+ * - with string argument: calculates average value of objects in array where
+ * property is a given string
```

Comment:
I don't understand the second condition

Reply:
@roman-mate please help us with english here

Reply:
* With no args it returns an average value of `items` in the array
* With string argument it returns an average value of `item[prop]`
* With function - it returns an average value of function results
* Functional argument is called with 3 args - an item, index, and an array itself

Reply:
positional arguments which have different meaning based on argument type is an anti-pattern. Let's use options instead. 

https://stackoverflow.com/questions/12826977/multiple-arguments-vs-options-object

Reply:
It is not about positional arguments but about [Function Polymorphism](https://www.oreilly.com/library/view/programming-javascript-applications/9781491950289/ch02.html)(search for 'Function Polymorphism' on the page please)

I wanted then to work with `typeof` operator

Reply:
It's a good idea to teach them typeof. 

Let's add 
* if options.accumulator is not a function, returns undefined.

And add a test case when we pass options.accumulator which is not a function. Then they practice typeof but we don't teach them antipatterns.

Reply:
Guys! @roman-mate @danmysak @mgrinko 
```
 * [1, 2, 6].calculateAverage() === 3
 *
 * [1, 2, 6].calculateAverage({propertyName: 'name'}) === undefined
 *
 * const users = [
 *  { id: 1, age: 10 },
 *  { id: 2, age: 20 },
 *  { id: 3, age: 60 },
 * ]
 *
 * users.calculateAverage({propertyName: 'id'}) === 2
 * users.calculateAverage({propertyName: 'name'}) === undefined
 * users.calculateAverage({
 *  accumulator: (user, index) => user.age + (index * 100)
 * }) === 130
 * users.calculateAverage({ accumulator: 'hello world' }) === undefined
```
wdyt?

Reply:
like it!

Reply:
Guys! @roman-mate @mgrinko @danmysak 
Will add also these test cases. I let them accept only `object` as an argument
```
 * [1, 2, 6].calculateAverage(['options', 'as', 'array']) === undefined
 * 
 * [1, 2, 6].calculateAverage(null) === undefined
 * 
 * [1, 2, 6].calculateAverage('options as string') === undefined
```

Reply:
also they should return `undefined` for empty object
```
* [1, 2, 6].calculateAverage({}) === undefined
```

Reply:
I wanted to keep it simple
That Why I think this extra conditions will add more confusion than profit

Reply:
let's mark it with *
It's a really interesting task. Btw I've implemented test cases so I don't want to lose the value :)